builtin/credential/userpass/path_users.go

```diff
@@ -95,43 +120,59 @@ func (b *backend) pathUserRead(
 	}, nil
 }
 
-func (b *backend) pathUserWrite(
-	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	name := strings.ToLower(d.Get("name").(string))
-	password := d.Get("password").(string)
-	policies := strings.Split(d.Get("policies").(string), ",")
-	for i, p := range policies {
-		policies[i] = strings.TrimSpace(p)
+func (b *backend) userCreateUpdate(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	username := strings.ToLower(d.Get("username").(string))
+	userEntry, err := b.user(req.Storage, username)
```

Comment:
In fact, that's what you're doing here. So you should just do this paradigm elsewhere and put in an explicit check for `""` into `b.user`.